mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -2864,6 +2843,9 @@ void TypeConverter::SignatureConversion::remapInput(unsigned origInputNo,
 
 LogicalResult TypeConverter::convertType(Type t,
                                          SmallVectorImpl<Type> &results) const {
+  assert(this && "expected non-null type converter");
```

Comment:
```suggestion
```
This assert only protects against a case that is already in UB as far as C++ is concerned and would e.g. get optimized out in release builds.